contrib/win32/install/install/components/c_server.wxs

```diff
@@ -0,0 +1,138 @@
+      <Component Id="cmp_sftpserver.exe">
+        <File Id="sftpserver.exe" KeyPath="yes" Source="$(var.SOURCEDIR)/sftp-server.exe" Name="sftp-server.exe" />
+      </Component>
+      <Component Id="cmp_sshshellhost.exe">
+        <File Id="sshshellhost.exe" KeyPath="yes" Source="$(var.SOURCEDIR)/ssh-shellhost.exe" Name="ssh-shellhost.exe" />
+      </Component>
+      <Component Id="cmp_opensshevents.man">
+        <File Id="opensshevents.man" KeyPath="yes" Source="$(var.SOURCEDIR)/openssh-events.man" Name="openssh-events.man">
+          <util:EventManifest />
+        </File>
+        <util:XmlFile
+          Id="xmlSetResourceFileName"
+          Sequence="1"
+          File="[#opensshevents.man]"
+          Action="setValue"
+          ElementPath="//*[\[]local-name() = 'provider'[\]]"
+          Name="resourceFileName"
+          Value="[#sshagent.exe]"
+          SelectionLanguage="XPath" />
+        <util:XmlFile
+          Id="xmlSetMessageFileName"
+          Sequence="2"
+          File="[#opensshevents.man]"
+          Action="setValue"
+          ElementPath="//*[\[]local-name() = 'provider'[\]]"
+          Name="messageFileName"
+          Value="[#sshagent.exe]"
+          SelectionLanguage="XPath" />
+      </Component>
+    </ComponentGroup>
+    <ComponentGroup Id="CG_SERVER_SSHAGENTSERVICE" Directory="INSTALLFOLDER">
+      <Component Id="cmp_sshagent.exe">
+        <File Id="sshagent.exe" Source="$(var.SOURCEDIR)/ssh-agent.exe" Name="ssh-agent.exe" KeyPath="yes" />
+        <ServiceInstall
+          Id="SVC_SSHAGENT"
+          Description="Agent to hold private keys used for public key authentication."
+          DisplayName="OpenSSH Authentication Agent"
+          ErrorControl="critical"
+          Interactive="no"
+          Name="ssh-agent"
+          Start="auto"
+          Type="ownProcess"
+          Vital="yes">
+          <util:ServiceConfig
+            ResetPeriodInDays="1"
+            FirstFailureActionType="restart"
+            SecondFailureActionType="restart"
+            ThirdFailureActionType="restart" />
+          <PermissionEx Sddl="D:(A;;CCLCSWRPWPDTLOCRRC;;;SY)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCLCSWLOCRRC;;;IU)(A;;CCLCSWLOCRRC;;;SU)(A;;RP;;;AU)" />
+        </ServiceInstall>
+        <ServiceControl
+          Id="SC_SSHAGENT_UNINSTALL"
+          Name="ssh-agent"
+          Stop="uninstall"
+          Remove="uninstall"
+          Wait="no" />
+      </Component>
+      <Component Id="cmp_server_sshagent_control" KeyPath="yes" Guid="5f300a9e-8896-4b3e-8b17-8ab1360b3ef1">
+        <Condition>
+          <![CDATA[NOT(DONT_START_SSHAGENT OR DONT_START_SERVICES)]]>
+        </Condition>
+        <ServiceControl
+          Id="SC_SSHAGENT_INSTALL"
+          Name="ssh-agent"
+          Start="install"
+          Wait="no" />
+      </Component>
+    </ComponentGroup>
+    <ComponentGroup Id="CG_SERVER_SSHDSERVICE" Directory="INSTALLFOLDER">
+      <Component Id="cmp_sshd.exe">
```

Comment:
Can you add a check to make sure the sshd.exe binary has "read & execute" permissions for the "authenticated users".
Not sure if you are setting the OpenSSH folder permissions explicitly. If not openssh bin folder inherits the folder permissions from it's parent folder.. If the parent folder doesn't have "read & execute" permissions for "authenticated users" then ssh connection will fail.